lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_check_beta_deps.rb

```diff
@@ -1,31 +1,56 @@
+require 'yaml'
+
 module Fastlane
   module Actions
     class IosCheckBetaDepsAction < Action
       def self.run(params)
         require_relative '../../helper/ios/ios_version_helper'
         require_relative '../../helper/ios/ios_git_helper'
 
-        beta_pods = []
-        File.open(params[:podfile]).each do |li|
-          beta_pods << li if li.match('^\s*\t*pod.*beta')
+        yaml = YAML.load_file(params[:lockfile])
+        non_stable_pods = {} # Key will be pod name, value will be reason for flagging
+
+        # Find pods referenced by commit and branch to a repo
+        yaml['EXTERNAL SOURCES'].each do |pod, options|
```

Comment:
I just realized that some `Podfile.lock` files might not have this section at all (if they don't have any external dependency), so better cover for the `nil` case.

(Ideally we should add a spec example to cover that too)
```suggestion
        yaml['EXTERNAL SOURCES']&.each do |pod, options|
```

Reply:
@spencertransier I've fixed this (and added test coverage for it) via dc5992fd2fbe7e4459b134ce6b1f82805254635a + b1a13d9b0fde57223dc12f28ef28b175d614c239, so PR is now ready for review again 🙇